pkg/v1/tkg/managementcomponents/management_component_install_test.go

```diff
@@ -169,3 +172,208 @@ var _ = Describe("Test WaitForManagementPackages", func() {
 		})
 	})
 })
+
+var _ = Describe("Test PauseAddonLifecycleManagement", func() {
+
+	type fakeGroupResource struct {
+		Group    string
+		Resource string
+	}
+
+	var (
+		clusterClient *fakes.ClusterClient
+		err           error
+		clusterName   string
+		addonName     string
+		namespace     string
+		notFoundError = apierrors.NewNotFound(
+			schema.GroupResource{"fakeGroup", "fakeGroupResource"},
+			"fakeGroupResource")
+	)
+
+	BeforeEach(func() {
+		clusterClient = &fakes.ClusterClient{}
+		clusterName = "mgmtCluster"
+		addonName = "addons-manager"
+	})
+
+	JustBeforeEach(func() {
+		err = PauseAddonLifecycleManagement(clusterName, addonName, namespace, clusterClient)
+	})
+
+	Context("Resource manipulation returns no errors", func() {
+		It("should return no error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+	Context("Patching resources returns unknown error", func() {
+		BeforeEach(func() {
+			clusterClient.PatchResourceReturns(errors.New("Unknown error"))
+		})
+		It("should return error", func() {
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("Patching resources returns not found", func() {
+		BeforeEach(func() {
+			clusterClient.PatchResourceReturns(notFoundError)
+		})
+		It("should return no error", func() {
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
But according your code, you had returned ```nil``` on ```NotFound``` return err value for ```PatchResource```. Wondering why error is expected here?